Automattic/Gravatar-SDK-Android
# `gravatar-ui`- First iteration of the Ghost UI

Closes #109 

### Description

We want the Gravatar UI components to have a "loading" state. That state should have a skeleton animation of 800ms.

SkeletonEffect contains the base code to be used along the different components to create their "loading" state.

All modified components can be used in two different ways:

- Passing a profile as a parameter (Previous way of working)
- Passing a UserProfileLoadingState. This way, we'll show a skeleton to indicate when the profile is loading. When the state changes to "Loaded", the composable will be updated to show the already loaded state.

<img width="433" alt="image" src="https://github.com/Automattic/Gravatar-SDK-Android/assets/6974554/5f347b45-f662-47b9-aeef-29a5f67286c4">


I'll apply the same pattern to the other components in the next PR.

### Testing Steps
- Run the different previews in a device
- Run the demo app and verify how the skeleton looks for the MiniProfileCard

https://github.com/Automattic/Gravatar-SDK-Android/assets/6974554/9bc63de8-523b-4ad7-9e06-26d4b5ecc3a7
